### PR TITLE
VZ-9977- Cluster API E2e test pipeline

### PR DESCRIPTION
**Pull request by theabmitra**
- New jenkins pipeline with CAPI tests 
- Deploys an OCNE cluster from KinD as an admin clusters 
- Deploys VZ dev profile on the new OCNE cluster 
- Cleans up the OCNE cluster

### CODE REVIEW

File: tests/e2e/backup/helpers/common_helper.go [hunk 1]
```diff
@@ -499,7 +499,7 @@ func Runner(bcmd *BashCommand, log *zap.SugaredLogger) *RunnerResponse {
 	bashCommand.Stdout = io.MultiWriter(os.Stdout, &stdoutBuf)
 	bashCommand.Stderr = io.MultiWriter(os.Stderr, &stderrBuf)
 
-	log.Infof("Executing command '%v'", bashCommand.String())
+	//log.Infof("Executing command '%v'", bashCommand.String())
```
Comment (michael-cico) [inline tests/e2e/backup/helpers/common_helper.go:502]: Should we remove this, or convert it to Debug?

File: tests/e2e/clusterapi/capi/capiModels.go [hunk 2]
```diff
@@ -0,0 +1,334 @@
+// Copyright (c) 2023, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package capi
+
+import "time"
+
+type Cluster struct {
```
Comment (michael-cico) [inline tests/e2e/clusterapi/capi/capiModels.go:8]: There's no API to reference?  Just wondering why we had to (re?) create these structures.

File: tests/e2e/clusterapi/capi/capi_test.go [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+	}
+	err := capiTest.UpdateOCINSGEW(ClusterName, "worker", "east-west traffic", &eastWestRule, t.Logs)
+	if err != nil {
+		return err
+	}
+
+	return capiTest.UpdateOCINSGEW(ClusterName, "control-plane", "east-west traffic", &eastWestRule, t.Logs)
+}
+
+/*
```
Comment (michael-cico) [inline tests/e2e/clusterapi/capi/capi_test.go:301]: dead code?

File: tests/e2e/clusterapi/capi/capi_test.go [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+			}, capiClusterCreationWaitTimeout, pollingInterval).Should(BeNil(), "Monitor Cluster Creation")
+		})
+
+		WhenClusterAPIInstalledIt("Create ClusterResourceSets on CAPI cluster", func() {
+			Eventually(func() error {
+				return capiTest.DeployClusterResourceSets(ClusterName, clusterResourceSetTemplate, t.Logs)
+			}, capiClusterCreationWaitTimeout, pollingInterval).Should(BeNil(), "Create CAPI cluster")
+		})
+
+		//WhenClusterAPIInstalledIt("Update NSG for new VCN created by CAPi cluster", func() {
```
Comment (michael-cico) [inline tests/e2e/clusterapi/capi/capi_test.go:332]: Dead code?

File: tests/e2e/clusterapi/capi/helpers.go [hunk 5]
```diff
@@ -0,0 +1,881 @@
...
+	if err != nil {
+		return err
+	}
+
+	log.Infof("----------- Pods running on workload cluster ---------------------")
+	return c.ShowPodInfo(client, clusterName, log)
+}
+
+/*
+func deleteNamespace(namespace string, log *zap.SugaredLogger) error {
```
Comment (michael-cico) [inline tests/e2e/clusterapi/capi/helpers.go:615]: dead code?

File: ci/clusterAPI/JenkinsfileCAPI [hunk 6]
```diff
@@ -0,0 +1,902 @@
...
+            saveConsoleScreenShots()
+            error "${err}"
+        }
+    }
+}
+
+
+def generateVerifyInstallStages(dumpRoot) {
+    return [
+        "verify-install keycloak": {
```
Comment (michael-cico) [inline ci/clusterAPI/JenkinsfileCAPI:591]: are we running all the `verify-install` tests, or just a subset?  If all, should we just run them as one `verify-install` call?

File: tests/e2e/clusterapi/capi/capi_test.go [hunk 7]
```diff
@@ -0,0 +1,432 @@
...
+	metrics.Emit(t.Metrics.With("undeployment_elapsed_time", time.Since(start).Milliseconds()))
+})
+
+var _ = AfterSuite(afterSuite)
+
+var t = framework.NewTestFramework("cluster-api")
+
+var capiTest = NewCapiTestClient()
+
+// 'It' Wrapper to only run spec if the Velero is supported on the current Verrazzano version
```
[the remark this answers is not in the document]
Comment (theabmitra) [reply tests/e2e/clusterapi/capi/capi_test.go:62]: done

File: tests/e2e/clusterapi/capi/capi_test.go (hunk 4 above)
Comment (theabmitra) [reply tests/e2e/clusterapi/capi/capi_test.go:332]: removed

File: tests/e2e/clusterapi/capi/capi_test.go (hunk 3 above)
Comment (theabmitra) [reply tests/e2e/clusterapi/capi/capi_test.go:301]: removed

File: ci/clusterAPI/JenkinsfileCAPI (hunk 6 above)
Comment (theabmitra) [reply ci/clusterAPI/JenkinsfileCAPI:591]: i have skipped prometheus verify , there are some issues there which i observed in the OLCNE pipeline as well. 
The other reason is to make these display in Jenkins

**Review by michael-cico — APPROVED** [review]

File: tests/e2e/clusterapi/capi/capiModels.go (hunk 2 above)
Comment (michael-cico) [reply tests/e2e/clusterapi/capi/capiModels.go:8]: OK, was just wondering if there was an Go API we could used directly, but sounds like it's either a PLS issue or there isn't one.

**Pull request merged**